### PR TITLE
Setup wizard

### PR DESCRIPTION
**Pull request by carise**
Pending:
 - [ ] Write config choices to deployment template
 - [ ] Give user the option to create a deployment
 - [ ] Save config to an actual setup config file (in case the user quits setup halfway; at least they can resume or make changes)
 - [ ] The actual installer wrapper script that someone can just `curl`

The code could probably be organized better too.

### CODE REVIEW

File: scripts/gcp_setup/environment/__init__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Copyright 2016 Google Inc.
```
Comment (mirons-google) [inline scripts/gcp_setup/environment/__init__.py]: 2017

File: scripts/gcp_setup/environment/gcloud_env.py [hunk 2]
```diff
@@ -0,0 +1,556 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Set up Forseti Security on GCP.
+
+This has been tested with python 2.7.
+
+TODO:
+ - Write config options to an output file and give option to read in.
```
Comment (mirons-google) [inline scripts/gcp_setup/environment/gcloud_env.py]: nit todo on one line.

File: scripts/gcp_setup/environment/gcloud_env.py [hunk 3]
```diff
@@ -0,0 +1,556 @@
...
+
+import json
+import re
+import subprocess
+import time
+
+from distutils.spawn import find_executable
+
+
+def org_id_from_name(org_name):
```
Comment (mirons-google) [inline scripts/gcp_setup/environment/gcloud_env.py]: nit: org_id_from_org_name(org_name)

Comment (mirons-google) [inline scripts/gcp_setup/environment/gcloud_env.py]: You'll want to include function doc here too.

Comment (mirons-google) [inline scripts/gcp_setup/environment/gcloud_env.py]: It'd be helpful to provide an example org_name so readers can understand.

File: scripts/gcp_setup/environment/gcloud_env.py [hunk 4]
```diff
@@ -0,0 +1,556 @@
...
+            stderr=subprocess.PIPE)
+        out, err = proc.communicate()
+        if proc.returncode != 0:
+            print(err)
+
+        if out:
+            self.auth_account = out.strip()
+
+    def list_organizations(self):
+        """Set the organization id."""
```
Comment (mirons-google) [inline scripts/gcp_setup/environment/gcloud_env.py]: nit: method name doesn't jive with the description.

File: scripts/gcp_setup/environment/gcloud_env.py [hunk 5]
```diff
@@ -0,0 +1,556 @@
...
+        Print link to go to GSuite service account and enable DWD, if GSuite
+        service account was created.
+        Print link for deployment manager dashboard.
+        """
+        self._print_banner('Post-setup instructions')
+
+        print('Congratulations! You\'ve completed the Forseti prerequisite '
+              'setup.')
+
+        if self.gsuite_service_acct:
```
Comment (mirons-google) [inline scripts/gcp_setup/environment/gcloud_env.py]: There's additional steps for them to enable the service-account and dwd-id in the GSuite control panel. I don't think I saw [those](http://forseti-security.io/service_accounts/) here.

File: scripts/gcp_setup/setup_project.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+"""
+
+import argparse
+
+from environment import gcloud_env
+
+def run():
+    """Run the steps for the gcloud setup."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--config', help='Path to setup config')
```
Comment (mirons-google) [inline scripts/gcp_setup/setup_project.py]: Should we include a sample config?

Comment (carise) [inline scripts/gcp_setup/setup_project.py]: I'm going to remove this for now

File: scripts/gcp_setup/environment/gcloud_env.py (hunk 5 above)
Comment (carise) [inline scripts/gcp_setup/environment/gcloud_env.py]: I'll just link to those steps.

**Comment by blueandgold** [comment]
Adding `cla: yes` label, as the PR author is a googler at the time of this PR was created.

**Pull request merged**